### PR TITLE
UI accelerators

### PR DESCRIPTION
**Pull request by Vorschreibung**
This change adds Accelerator key functionality to ET's Menus, demo: https://www.youtube.com/watch?v=j33ZnEXoO54 by parsing `^_` metacharacters out of UI texts, highlighting them and registering their action to the onKey handler.

If the general idea of Accelerators are desirable and the current implementation suffices, I'd follow up filling in more Accelerators for UI Elements until it's in a release-able state.

### CODE REVIEW

File: src/ui/ui_main.c [hunk 1]
```diff
@@ -250,6 +251,26 @@ void _UI_DrawRect(float x, float y, float width, float height, float size, const
...
+void _UI_DrawBox(float x, float y, float width, float height, const float *color)
+{
+	trap_R_SetColor(color);
+
+	UI_AdjustFrom640(&x, &y, &width, &height);
+	trap_R_DrawStretchPic(x, y, width, height, 0, 0, 0, 0, uiInfo.uiDC.whiteShader);
+
+	trap_R_SetColor(NULL);
+}
+
```
Comment (Aciz) [inline src/ui/ui_main.c]: `UI_FillRect` (`DC->fillRect`) is line by line identical to this.

https://github.com/etlegacy/etlegacy/blob/ac4b2deeccd6cdbb9cea4e6ba87400dbf602ab87/src/ui/ui_atoms.c#L309-L317

Comment (Vorschreibung) [reply src/ui/ui_main.c]: Total oversight, thanks! Fixed

**Comment by jackeri** [comment]
Really like this idea, though it does break the automatic generation of the translation files (which we haven't really used in a long time), but if that is all we can probably live with it.

File: src/ui/ui_menu.c [hunk 2]
```diff
@@ -914,6 +914,49 @@ void  Menus_Activate(menuDef_t *menu)
...
+		const char *s   = item->text;
+
+		if (s == NULL) {
+			continue;
+		}
+
+		int       len                  = Q_UTF8_Strlen(s);
+		int       count                = 0;
+		short int accelerator          = 0;
+		qboolean  accelerator_upcoming = qfalse;
```
Comment (Aranud) [inline src/ui/ui_menu.c]: This doesn't respect C89 standard. We try to stick as much as possible to this declaration style for concistency with code. Variables should be declared at top of the scope then `len` assigned after `s` NULL check.

**Review by Aranud — CHANGES_REQUESTED** [review]

File: src/ui/ui_menu.c (hunk 2 above)
Comment (Vorschreibung) [reply src/ui/ui_menu.c]: :metal: Done

**Review by Aranud — APPROVED** [review]

**Comment by Vorschreibung** [comment]
The scope of this has never been sufficiently clarified while meta-chars to translation files were explicitly rejected. - I think it makes more sense to close this altogether as that would constitute a different approach altogether that someone can take up in the future and where it's unlikely these commits would be of any help.

**Pull request closed without merging**